### DISCUSSION TITLE
Why does Forward TTS uses Interpolating in pitch feature?

### DISCUSSION THREAD

**Discussion by amitaie** [coqui-ai/TTS#1997] [General Q&A]
Hey, Saw that FastSpeech2 and others use Interpolation in the Pitch extractor. and i don't understand what for.
if anyone knows the idea behind it i would love to know.
thank you in advanced.

**Comment by p0p4k** [comment]
Can you link me the paper/codes and exact page/line numbers? I will try to read them and get back to you

**Reply by amitaie** [reply]
Sure,
In FastSpeech2 paper they mention it (page 14, C.2):

> `First we extract the pitch contour using PyWorldVocoder10. Since CWT
> is very sensitive to discontinuous signals, we preprocess the pitch contour as follows: 1) we use linear interpolation to fill the unvoiced frame
> in pitch contour; 2) we transform the resulting pitch contour to logarithmic scale; 3) we normalize it to zero mean and unit variance for each
> utterance, and we have to save the original utterance-level mean and variance for pitch contour reconstruction; and 4) we convert the normalized
> pitch contour to pitch spectrogram using continuous wavelet transform
> following Equation 1.`

They do say it is for continuous signal but I don't understand how does it work, they change the pitch in a way that the model won't know when it's unvoice or not,

**Reply by erogol** [reply]
We follow what FastPitch does in terms of pitch processing.
